### PR TITLE
TM-1093, #317 fixed

### PR DESCRIPTION
**Pull request by ArekZc**

### CODE REVIEW

File: src/containerjs/handler/tags.js [hunk 1]
```diff
@@ -86,7 +86,16 @@
 
                     }
 
-                    delayCallback(enableDelay, state.dataLayer);
+                    // Fix attribute listening on events
+                    if (
+                        state.dataLayer.element.href !== "" ||
```
Comment (kallosz) [inline src/containerjs/handler/tags.js]: move this condition to separate method

Comment (kallosz) [inline src/containerjs/handler/tags.js]: revert condition to avoid increasing complexity

Comment (kallosz) [inline src/containerjs/handler/tags.js]: this statement should be before `if ($debugger.breakpoints.isEnabled()) {` statement

**Comment by kallosz** [comment]
Missing unit test.

**Comment by kallosz** [comment]
@ArekZc wrong branch destination. This should be hotfix to master. We decide with @pszostak to release this ASAP in 1.1.1 version.

**Comment by ArekZc** [comment]
@kallosz This is general loop of application, so it need to be tested by integration not unit, I create ticket for 1.3 version to refactor this loop to send them to separate services. Now we have problem with integration tests of container

File: src/containerjs/handler/tags.js [hunk 2]
```diff
@@ -80,13 +85,23 @@
...
                     }
 
-                    delayCallback(enableDelay, state.dataLayer);
+                    element = state.dataLayer.element;
+                    hasClickEventAttached = (element.onclick !== undefined) ? element.onclick !== null : false;
+                    hasSubmitEventAttached = (element.onclick !== undefined) ? element.onsubmit !== null : false;
+                    hasHrefAttribute = element.href !== "";
+
+                    if (
+                        hasHrefAttribute ||
```
Comment (kallosz) [inline src/containerjs/handler/tags.js]: G28: Encapsulate Conditionals

Boolean logic is hard enough to understand without having to see it in the context of an if
or while statement. Extract functions that explain the intent of the conditional.
For example:
`if (shouldBeDeleted(timer))`
is preferable to
`if (timer.hasExpired() && !timer.isRecurrent())`

Comment (kallosz) [inline src/containerjs/handler/tags.js]: if you move this to separate method you will be able to test this in unit test also.

**Comment by ArekZc** [comment]
@kallosz We discuss this with @pszostak. We decide to done work with refactoring this service in next sprint, now it's hotfix, so users could use "onclick" attribute to block link forwarding

File: src/containerjs/handler/tags.js (hunk 2 above)
Comment (ArekZc) [inline src/containerjs/handler/tags.js]: Yes, I know that much logic in if statement is not readable, so i change them to variables, but when you move them to separate method you will break hermetization by expose private logic of service

**Pull request merged**